### PR TITLE
EES-2614 Unit tests for Publication#LatestRelease and #LatestPublishe…

### PR DESCRIPTION
**Pull request by markhiveit**
…dReleasse

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            {
+                Releases = new List<Release>
+                {
+                    new()
+                    {
+                        Id = release2000Id,
+                        Version = 0,
+                        ReleaseName = "2000",
+                        TimePeriodCoverage = TimeIdentifier.AcademicYear,
+                        Published = DateTime.UtcNow.AddDays(-2),
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs:74]: Published value doesn't matter here as long as they are published, and setting up one with -2 and one with -1 implies you might be expecting it to influence the order in some way. I'd be tempted to just use `DateTime.UtcNow` as it's the Year and TimePeriod and presence of the Published value which are important.

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                        Published = null,
+                        TimePeriodCoverage = TimeIdentifier.AcademicYear,
+                        ApprovalStatus = ReleaseApprovalStatus.Draft,
+                    },
+                },
+            };
+
+            var result = publication.LatestPublishedRelease();
+            Assert.Equal(release2001Id, result.Id);
+        }
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs:101]: I think you could also add tests for LatestPublished_NoPublishedRelease and LatestPublished_NoReleases where we'd expect to catch bugs if we accidently assume at least one Release exists.

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                        TimePeriodCoverage = TimeIdentifier.AcademicYear,
+                        ApprovalStatus = ReleaseApprovalStatus.Draft,
+                    },
+                },
+            };
+
+            var result = publication.LatestRelease();
+            Assert.Equal(release2001AmendmentId, result.Id);
+        }
+    }
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs:159]: Like above I think you could add a `LatestRelease_NoRelease`. Has Publication.cs enabled nullable reference types yet? If so should be clearer to see that we can expect both these methods to return null.

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            var publication = new Publication
+            {
+                Releases = new List<Release>
+                {
+                    new()
+                    {
+                        Id = release2000Id,
+                        Version = 0,
+                        ReleaseName = "2000",
+                        TimePeriodCoverage = TimeIdentifier.AcademicYear,
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs:119]: You could use a static import for the TimeIdentifiers and ReleaseApprovalStatus

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                        Published = DateTime.UtcNow.AddDays(-1),
+                        ApprovalStatus = ReleaseApprovalStatus.Approved,
+                    },
+                    new()
+                    {
+                        Id = release2001Id,
+                        Version = 0,
+                        ReleaseName = "2001",
+                        Published = DateTime.UtcNow.AddDays(-1),
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs:139]: Same comment about Published date/time as above. If they were all published on the same day we should expect the same result, but the data is setting up days -1 and -2 which made me think it's influencing the order in some way.

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs (hunk 3 above)
Comment (markhiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/PublicationTests.cs:200]: I've enabled nullable to Publication.cs and PublicationTests.cs. I had to make various changes to Publication.cs to make sure no changes appear if you make a migration - which I tested and a blank migration was created.

**Review by duncan-at-hiveit — APPROVED** [review]

**Pull request merged**